### ISSUE TITLE
Various Mac improvements

### ISSUE THREAD

**Issue by bryceco** [Keriew/augustus#559]
I'm playing on Mac and there are several things I wish behaved differently. I'm happy to implement these changes and submit a pull request but I'd like to know whether they'd be welcome:

* When the window is no longer foreground I'd like the game to pause and the sound to stop. For example if I'm in the middle of playing a game and want to watch a YouTube video of GamerZakh on that mission it's cumbersome to have to disable/enable audio to do so. I looked through the SDL API a bit and it looks like there might be a way to hook into UIApplicationDelegate in order to detect state transfers but I'm not sure.

* When exiting via Command-Q it would be really nice for the game to save automatically (to something like "Current Game.svx") and then automatically load it the next time you start. This would be more similar to how other Mac apps behave.

* The game continues running when I close the lid of my laptop. This is obvious based on the heat generated, and means that I'll kill my battery if I don't proactively quit the app. I notice that in Activity Monitor the "Preventing Sleep" column is marked as "Yes" for Augustus. The SDL docs say "The screensaver is disabled by default since SDL 2.0.2" so I tried adding a call to SDL_EnableScreenSaver() during initialization but it didn't change anything. Any ideas?

**Comment by crudelios** [comment]
Hi! I'm finally taking the time to answer your comment.


> * When the window is no longer foreground I'd like the game to pause and the sound to stop. For example if I'm in the middle of playing a game and want to watch a YouTube video of GamerZakh on that mission it's cumbersome to have to disable/enable audio to do so. I looked through the SDL API a bit and it looks like there might be a way to hook into UIApplicationDelegate in order to detect state transfers but I'm not sure.

This has been tried in Julius, without success. The problem is that we need consistent behaviour across all platforms and SDL events are not consistent between platforms regarding what events get triggered when a window becomes active/inactive.

TL;DR: Even if you manage to make it work for mac, it'll likely break things on other platforms.


> * When exiting via Command-Q it would be really nice for the game to save automatically (to something like "Current Game.svx") and then automatically load it the next time you start. This would be more similar to how other Mac apps behave.

This would be inconsistent with all other platforms and as mentioned above we want consistency. Although making it an option would work!


> * The game continues running when I close the lid of my laptop. This is obvious based on the heat generated, and means that I'll kill my battery if I don't proactively quit the app. I notice that in Activity Monitor the "Preventing Sleep" column is marked as "Yes" for Augustus. The SDL docs say "The screensaver is disabled by default since SDL 2.0.2" so I tried adding a call to SDL_EnableScreenSaver() during initialization but it didn't change anything. Any ideas?

I have no idea regarding this one. Maybe @bvschaik knows?

**Comment by crudelios** [comment]
Closing this due to inactivity
